### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by tlx02**
Combined the 2 methods - setTask and replaceTask since they are similar.
Changed mark and unmark command to throw error when the task is already marked/unmarked.

### CODE REVIEW

**Review by phualiting — APPROVED** [review]
LGTM

**Review by dlimyy — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/task/DistinctTaskList.java [hunk 1]
```diff
@@ -55,43 +55,34 @@ public void addTask(Task taskAdded) {
...
             throw new TaskNotFoundException();
         }
-        if (contains(editedTask) && !editedTask.isSameTask(target)) {
+
+        if (isSameTask && !target.isSameTask(editedTask)) {
+            throw new WrongTaskModifiedException();
+        }
+
+        boolean isDuplicateTask = contains(editedTask) && !editedTask.isSameTask(target);
+        if (!isSameTask && contains(editedTask) && isDuplicateTask) {
```
Comment (dlimyy) [inline src/main/java/seedu/address/model/task/DistinctTaskList.java:82]: Spotted minor bug in code after merging where contains(ediedTask) here is unnecessary

Comment (tlx02) [reply src/main/java/seedu/address/model/task/DistinctTaskList.java:82]: Thanks for spotting the bug, I have removed the extra 'contains'.

**Pull request merged**